spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/LocalizedQueueConnectionFactoryTests.java

```diff
@@ -90,22 +86,25 @@ public void testFailOver() throws Exception {
 		String username = "guest";
 		String password = "guest";
 		final AtomicBoolean firstServer = new AtomicBoolean(true);
+		final Client client1 = doCreateClient(adminUris[0], username, password, nodes[0]);
+		final Client client2 = doCreateClient(adminUris[1], username, password, nodes[1]);
+		final Map<String, ConnectionFactory> mockCFs = new HashMap<String, ConnectionFactory>();
+		CountDownLatch latch1 = new CountDownLatch(1);
+		CountDownLatch latch2 = new CountDownLatch(1);
+		mockCFs.put(rabbit1, mockCF(rabbit1, latch1));
+		mockCFs.put(rabbit2, mockCF(rabbit2, latch2));
 		LocalizedQueueConnectionFactory lqcf = new LocalizedQueueConnectionFactory(defaultConnectionFactory, addresses,
 				adminUris, nodes, vhost, username, password, false, null) {
 
-			private final String[] nodes = new String[] { "rabbit@foo", "rabbit@bar" };
-
-
 			@Override
 			protected Client createClient(String adminUri, String username, String password)
 					throws MalformedURLException, URISyntaxException {
-				return doCreateClient(adminUri, username, password, firstServer.get() ? nodes[0] : nodes[1]);
+				return firstServer.get() ? client1 : client2;
 			}
 
-
 			@Override
 			protected ConnectionFactory createConnectionFactory(String address, String node) throws Exception {
-				return mockCF(address);
```

Comment:
This is the root of cause...
Yeah, need to take into account that any mock must be created before the actual test work.
Feel free to merge, if we don't have such a vulnerability somewhere else :smile: